src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/pbs/PBSCommand.groovy

```diff
@@ -146,37 +124,28 @@ class PBSCommand extends Command {
         String accountName = job.customUserAccount ?: parentJobManager.getUserAccount()
         boolean holdJobsOnStart = parentJobManager.isHoldJobsEnabled()
 
+        List<String> parameters = []
+        parameters << ("${PARM_JOBNAME} ${id}" as String)
+        if (holdJobsOnStart) parameters << "-h"
+        if (accountName) parameters << ("${PARM_ACCOUNT} ${accountName}" as String)
+        parameters << getAdditionalCommandParameters()
+        parameters << ("-w ${getWorkingDirectory()}" as String)
+        parameters << getLoggingParameter(job.jobLog)
+        if (email) parameters << getEmailParameter(email)
+        if (groupList && groupList != "UNDEFINED") parameters << getGroupListString(groupList)
+        if (umask) parameters << getUmaskString(umask)
+        parameters << assembleProcessingCommands()
+        parameters << assembleDependencyString()
+        parameters << assembleVariableExportString()
+
         StringBuilder qsubCall = new StringBuilder(EMPTY)
 
         if (job.getToolScript()) {
             qsubCall << "echo " << escapeBash(job.getToolScript()) << " | "
         }
 
-        qsubCall << QSUB << PARM_JOBNAME << id
-
-        if (holdJobsOnStart) qsubCall << " -h "
-
-        if (accountName) qsubCall << PARM_ACCOUNT << accountName << " "
-
-        qsubCall << getAdditionalCommandParameters()
-
-        if (loggingDirectory) qsubCall << PARM_LOGPATH << loggingDirectory
-
-        qsubCall << getJoinLogParameter()
-
-        if (isArray) qsubCall << assembleArraySettings()
-
-        if (email) qsubCall << getEmailParameter(email)
-
-        if (groupList && groupList != "UNDEFINED") qsubCall << getGroupListString(groupList)
-
-        if (umask) qsubCall << getUmaskString(umask)
-
-        qsubCall << assembleProcessingCommands()
-
-        qsubCall << assembleDependencyString()
-
-        qsubCall << assembleVariableExportString()
+        qsubCall << QSUB
```

Comment:
Same comment as above for the LSF code. The order in the code does not align with the order in the output.

Reply:
same

Reply:
As you touched all the files, could you also unify the code? It is the (nearly) the same for PBS/SGE/SLURM and LSF. Think it should be quite easy.

Reply:
Done